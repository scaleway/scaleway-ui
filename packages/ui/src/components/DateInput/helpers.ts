import { CURRENT_MONTH, CURRENT_YEAR } from './constants'

// First day of the month for a given year
export const getMonthFirstDay = (
  month = CURRENT_MONTH,
  year = CURRENT_YEAR,
) => {
  const firstDay = new Date(year, month - 1, 1).getDay()

  // Change so that a week starts on monday
  return firstDay === 0 ? 6 : firstDay - 1
}

export const addZero = (value: number) => `${value}`.padStart(2, '0')

export const getPreviousMonth = (month: number, year: number) => {
  if (month === 1) {
    return [12, year - 1]
  }

  return [month - 1, year]
}

export const getNextMonth = (month: number, year: number) => {
  if (month === 12) {
    return [1, year + 1]
  }

  return [month + 1, year]
}

// Checks if two date values are of the same month and year
export const isSameMonth = (date: Date, basedate = new Date()) =>
  basedate.getMonth() === date.getMonth() &&
  basedate.getFullYear() === date.getFullYear()

// (bool) Checks if two date values are the same day
export const isSameDay = (date: Date, basedate = new Date()) =>
  basedate.getDate() === date.getDate() &&
  basedate.getMonth() + 1 === date.getMonth() + 1 &&
  basedate.getFullYear() === date.getFullYear()

// Default format if none is provided
const getDateISO = (showMonthYearPicker: boolean, date?: Date) => {
  if (date) {
    if (showMonthYearPicker) {
      return [addZero(date.getMonth() + 1), date.getFullYear()].join('/')
    }

    return [
      addZero(date.getMonth() + 1),
      addZero(date.getDate()),
      date.getFullYear(),
    ].join('/')
  }

  return ''
}

export const formatValue = (
  computedValue: Date | null,
  computedRange: {
    start: Date | null
    end: Date | null
  } | null,
  showMonthYearPicker: boolean,
  selectsRange: boolean,
  format?: (value?: Date) => string | undefined,
) => {
  if (selectsRange && computedRange) {
    return format
      ? `${format(computedRange.start ?? undefined) ? `${format(computedRange.start ?? undefined)} - ` : ''}${format(computedRange.end ?? undefined) ?? ''}`
      : `${getDateISO(showMonthYearPicker, computedRange.start ?? undefined)}${computedRange.start ? ' - ' : ''}${getDateISO(showMonthYearPicker, computedRange.end ?? undefined)}`
  }

  if (computedValue && format) {
    return format(computedValue)
  }
  if (computedValue) {
    return getDateISO(showMonthYearPicker, computedValue)
  }

  return undefined
}
