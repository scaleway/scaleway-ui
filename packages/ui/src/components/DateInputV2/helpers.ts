export const POPUP_WIDTH = '264px'

// The current year
export const CURRENT_YEAR = new Date().getFullYear()

// The current month starting from 1 - 12
// 1 => January, 12 => December
export const CURRENT_MONTH = new Date().getMonth() + 1

// The current day 1-28/29/30/31

// Weeks displayed on calendar
export const CALENDAR_WEEKS = 6

// Number days in a month for a given year
export const getMonthDays = (month = CURRENT_MONTH, year = CURRENT_YEAR) => {
  const months30 = [4, 6, 9, 11] // 30 days in april, june, september and november
  const leapYear = year % 4 === 0

  if (month === 2) {
    return leapYear ? 29 : 28 // Days in february
  }

  return months30.includes(month) ? 30 : 31
}

// First day of the month for a given year from 1 (monday) - 7 (saturday)
export const getMonthFirstDay = (month = CURRENT_MONTH, year = CURRENT_YEAR) =>
  new Date(`${year}-${month}-01`).getDay() + 1

// Adds a zero for 1-digit months/days
export const addZero = (value: number) => `${value}`.padStart(2, '0')

export const getPreviousMonth = (month: number, year: number) => {
  if (month === 1) {
    return [12, year - 1]
  }

  return [month - 1, year]
}

export const getNextMonth = (month: number, year: number) => {
  if (month === 12) {
    return [1, year + 1]
  }

  return [month + 1, year]
}

// Checks if two date values are of the same month and year
export const isSameMonth = (date: Date, basedate = new Date()) =>
  basedate.getMonth() + 1 === date.getMonth() + 1 &&
  basedate.getFullYear() === date.getFullYear()

// (bool) Checks if two date values are the same day
export const isSameDay = (date: Date, basedate = new Date()) =>
  basedate.getDate() === date.getDate() &&
  basedate.getMonth() + 1 === date.getMonth() + 1 &&
  basedate.getFullYear() === date.getFullYear()
