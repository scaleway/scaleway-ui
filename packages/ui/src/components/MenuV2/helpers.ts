import { Children, type ReactNode, isValidElement } from 'react'

/**
 * Search inside a children (React Element) recursively until a result is found
 */
export const searchChildren = (
  children: ReactNode,
  searchString: string,
): ReactNode[] => {
  const matches: ReactNode[] = []
  const searchRegex = new RegExp(searchString, 'i')

  Children.forEach(children, child => {
    if (typeof child === 'string' && child.match(searchRegex)) {
      matches.push(child)
    } else if (isValidElement(child)) {
      const childProps = child.props as { children: ReactNode }

      // Recursively search the children of this element
      const childMatches = searchChildren(childProps.children, searchString)

      if (childMatches.length > 0) {
        // If any matches are found within this child's children, push the entire child element
        matches.push(child)
      }
    }
  })

  return matches
}
