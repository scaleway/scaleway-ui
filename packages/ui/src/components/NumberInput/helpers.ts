export const bounded = (value: number, min: number, max: number) =>
  Math.max(min, Math.min(value, max))

export const roundStep = (value: number, step: number, direction: number) =>
  direction === -1
    ? Math.floor(value / step) * step
    : Math.ceil(value / step) * step

export const getMinusRoundedValue = (currentValue: number, step: number) =>
  currentValue % step === 0
    ? roundStep(currentValue - step, step, -1)
    : roundStep(currentValue, step, -1)

export const getPlusRoundedValue = (currentValue: number, step: number) =>
  currentValue % step === 0
    ? roundStep(currentValue + step, step, 1)
    : roundStep(currentValue, step, 1)
