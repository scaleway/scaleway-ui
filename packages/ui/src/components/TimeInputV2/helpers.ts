const format = (value: string, timeFormat: 12 | 24, isHour?: boolean) => {
  if (value === '') return ''
  if (timeFormat === 12 && isHour) {
    const formattedValue = Number.parseInt(value, 10) % 12
    const finalValue = formattedValue === 0 ? 12 : formattedValue

    return finalValue.toString().padStart(2, '0')
  }

  return value.padStart(2, '0')
}

const computePeriod = (hours?: string, period?: 'am' | 'pm') => {
  if (period) return period

  if (hours && Number.parseInt(hours, 10) >= 12) return 'pm'

  return 'am'
}

export const isNumber = (key: string) => /^\d+$/.test(key)
export const isAOrP = (key: string) => ['a', 'A', 'p', 'P'].includes(key)

const isValidNewHour = (old: string, key: string, timeFormat: number) => {
  const newValue = Number.parseInt(old + key, 10)

  return (
    (timeFormat === 24 && newValue < 24) || (timeFormat === 12 && newValue < 13)
  )
}

export const isCompleteHour = (timeFormat: number, value?: string) => {
  const computedValue = Number.parseInt(value ?? '0', 10)

  if (timeFormat === 12 && computedValue > 1) return true
  if (timeFormat === 24 && computedValue > 2) return true

  return false
}

export const canConcat = (
  oldValue: string | undefined,
  type: 'h' | 'm' | 's',
  key: string,
  timeFormat: 24 | 12,
) =>
  oldValue &&
  ((type === 'h' && oldValue && isValidNewHour(oldValue, key, timeFormat)) ||
    (type !== 'h' && Number.parseInt(oldValue, 10) < 6))

export const formatValue = (
  value: {
    h: string
    m: string
    s: string
    period?: string
  },
  timeFormat: 24 | 12,
) => ({
  h: format(value.h, timeFormat, true),
  m: format(value.m, timeFormat),
  s: format(value.s, timeFormat),
  period: computePeriod(
    value.h,
    value.period === 'am' || value.period === 'pm' ? value.period : undefined,
  ),
})
