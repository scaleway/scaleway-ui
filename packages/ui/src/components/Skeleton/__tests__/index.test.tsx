import { Skeleton, skeletonTypes } from '..'
import { shouldMatchEmotionSnapshot } from '../../../../.jest/helpers'

describe('Skeleton', () => {
  test('renders default variant', () =>
    shouldMatchEmotionSnapshot(<Skeleton />))

  test.each(skeletonTypes)('renders correctly with type="%s"', type =>
    shouldMatchEmotionSnapshot(<Skeleton variant={type} />),
  )
})
