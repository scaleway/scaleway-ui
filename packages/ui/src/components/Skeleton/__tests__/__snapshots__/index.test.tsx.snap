// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton renders correctly with type="block" 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    left: -25%;
  }

  to {
    left: 100%;
  }
}

.css-bgbcvp-StyledContainer {
  position: relative;
  display: block;
  width: 100%;
  overflow: hidden;
  cursor: progress;
}

.css-1nd3dqf-StyledList {
  min-height: 200px;
  width: 100%;
  height: 100%;
  padding: 16px;
  border: 1px solid #d4dae7;
  border-radius: 4px;
  margin: 0;
}

.css-1nd3dqf-StyledList>.eomzb311:not(:last-child) {
  border-bottom: 1px solid #e6e9f0;
}

.css-1udri31-StyledLine {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 24px 16px;
}

.css-115kc0m-IconSkeleton {
  margin-right: 8px;
  width: 32px;
  height: 32px;
  min-width: 32px;
  border-radius: 8px;
  background-color: #d4dae7;
}

.css-1gmiakp-Line {
  height: 12px;
  width: 120px;
  max-width: 100%;
  border-radius: 8px;
  background-color: #d4dae7;
}

.css-q3qai6-StyledDiv {
  position: absolute;
  top: 0;
  height: 100%;
  width: 25%;
  opacity: 0.8;
  background: linear-gradient(
      90deg,
      #ffffff00,
      #ffffff4D,
      #ffffff66,
      #ffffff4D,
      #ffffff00
    );
  -webkit-animation: animation-0 1s linear infinite;
  animation: animation-0 1s linear infinite;
  -webkit-animation-direction: alternate;
  animation-direction: alternate;
}

@media (prefers-reduced-motion: reduce) {
  .css-q3qai6-StyledDiv {
    -webkit-animation: unset;
    animation: unset;
  }
}

<div
    class="css-bgbcvp-StyledContainer e181dehd1"
  >
    <ul
      class="css-1nd3dqf-StyledList eomzb310"
    >
      <li
        class="css-1udri31-StyledLine eomzb311"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </li>
      <li
        class="css-1udri31-StyledLine eomzb311"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </li>
      <li
        class="css-1udri31-StyledLine eomzb311"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </li>
    </ul>
    <div
      class="css-q3qai6-StyledDiv e181dehd0"
    />
  </div>
</DocumentFragment>
`;

exports[`Skeleton renders correctly with type="blocks" 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    left: -25%;
  }

  to {
    left: 100%;
  }
}

.css-bgbcvp-StyledContainer {
  position: relative;
  display: block;
  width: 100%;
  overflow: hidden;
  cursor: progress;
}

.css-sprtbs-StyledContainer {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: 16px;
}

.css-2szx0p-Block {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 16px;
  border: 1px solid #d4dae7;
  border-radius: 4px;
  overflow: hidden;
}

.css-115kc0m-IconSkeleton {
  margin-right: 8px;
  width: 32px;
  height: 32px;
  min-width: 32px;
  border-radius: 8px;
  background-color: #d4dae7;
}

.css-1gmiakp-Line {
  height: 12px;
  width: 120px;
  max-width: 100%;
  border-radius: 8px;
  background-color: #d4dae7;
}

.css-q3qai6-StyledDiv {
  position: absolute;
  top: 0;
  height: 100%;
  width: 25%;
  opacity: 0.8;
  background: linear-gradient(
      90deg,
      #ffffff00,
      #ffffff4D,
      #ffffff66,
      #ffffff4D,
      #ffffff00
    );
  -webkit-animation: animation-0 1s linear infinite;
  animation: animation-0 1s linear infinite;
  -webkit-animation-direction: alternate;
  animation-direction: alternate;
}

@media (prefers-reduced-motion: reduce) {
  .css-q3qai6-StyledDiv {
    -webkit-animation: unset;
    animation: unset;
  }
}

<div
    class="css-bgbcvp-StyledContainer e181dehd1"
  >
    <div
      class="css-sprtbs-StyledContainer ellyxki1"
    >
      <div
        class="css-2szx0p-Block ellyxki0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-2szx0p-Block ellyxki0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-2szx0p-Block ellyxki0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-2szx0p-Block ellyxki0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-2szx0p-Block ellyxki0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-2szx0p-Block ellyxki0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-2szx0p-Block ellyxki0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-2szx0p-Block ellyxki0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
    </div>
    <div
      class="css-q3qai6-StyledDiv e181dehd0"
    />
  </div>
</DocumentFragment>
`;

exports[`Skeleton renders correctly with type="box" 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    left: -25%;
  }

  to {
    left: 100%;
  }
}

.css-bgbcvp-StyledContainer {
  position: relative;
  display: block;
  width: 100%;
  overflow: hidden;
  cursor: progress;
}

.css-sprtbs-StyledContainer {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: 16px;
}

.css-1eti1qk-Block {
  height: 130px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -ms-flex-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  justify-content: space-evenly;
  border: 1px solid #d4dae7;
  border-radius: 4px;
  overflow: hidden;
}

.css-115kc0m-IconSkeleton {
  margin-right: 8px;
  width: 32px;
  height: 32px;
  min-width: 32px;
  border-radius: 8px;
  background-color: #d4dae7;
}

.css-1gmiakp-Line {
  height: 12px;
  width: 120px;
  max-width: 100%;
  border-radius: 8px;
  background-color: #d4dae7;
}

.css-q3qai6-StyledDiv {
  position: absolute;
  top: 0;
  height: 100%;
  width: 25%;
  opacity: 0.8;
  background: linear-gradient(
      90deg,
      #ffffff00,
      #ffffff4D,
      #ffffff66,
      #ffffff4D,
      #ffffff00
    );
  -webkit-animation: animation-0 1s linear infinite;
  animation: animation-0 1s linear infinite;
  -webkit-animation-direction: alternate;
  animation-direction: alternate;
}

@media (prefers-reduced-motion: reduce) {
  .css-q3qai6-StyledDiv {
    -webkit-animation: unset;
    animation: unset;
  }
}

<div
    class="css-bgbcvp-StyledContainer e181dehd1"
  >
    <div
      class="css-sprtbs-StyledContainer ev9nufa1"
    >
      <div
        class="css-1eti1qk-Block ev9nufa0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-1eti1qk-Block ev9nufa0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-1eti1qk-Block ev9nufa0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-1eti1qk-Block ev9nufa0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-1eti1qk-Block ev9nufa0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-1eti1qk-Block ev9nufa0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-1eti1qk-Block ev9nufa0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-1eti1qk-Block ev9nufa0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
    </div>
    <div
      class="css-q3qai6-StyledDiv e181dehd0"
    />
  </div>
</DocumentFragment>
`;

exports[`Skeleton renders correctly with type="donut" 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    left: -25%;
  }

  to {
    left: 100%;
  }
}

.css-bgbcvp-StyledContainer {
  position: relative;
  display: block;
  width: 100%;
  overflow: hidden;
  cursor: progress;
}

.css-a1981g-StyledContainer {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.css-1mvslid-StyledSVG {
  height: 206px;
  width: 206px;
  -webkit-transform: rotate(-90deg);
  -moz-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.css-1ebk3po-StyledCircle {
  transform-origin: 50% 50%;
  stroke: #d4dae7;
  stroke-width: 18;
  stroke-linecap: butt;
  fill: none;
}

.css-1a6m9ou-LineList {
  display: grid;
  list-style: none;
  gap: 16px;
}

.css-1gmiakp-Line {
  height: 12px;
  width: 120px;
  max-width: 100%;
  border-radius: 8px;
  background-color: #d4dae7;
}

.css-q3qai6-StyledDiv {
  position: absolute;
  top: 0;
  height: 100%;
  width: 25%;
  opacity: 0.8;
  background: linear-gradient(
      90deg,
      #ffffff00,
      #ffffff4D,
      #ffffff66,
      #ffffff4D,
      #ffffff00
    );
  -webkit-animation: animation-0 1s linear infinite;
  animation: animation-0 1s linear infinite;
  -webkit-animation-direction: alternate;
  animation-direction: alternate;
}

@media (prefers-reduced-motion: reduce) {
  .css-q3qai6-StyledDiv {
    -webkit-animation: unset;
    animation: unset;
  }
}

<div
    class="css-bgbcvp-StyledContainer e181dehd1"
  >
    <div
      class="css-a1981g-StyledContainer exw3d033"
    >
      <svg
        class="css-1mvslid-StyledSVG exw3d032"
      >
        <circle
          class="css-1ebk3po-StyledCircle exw3d031"
          cx="103"
          cy="103"
          r="90"
        />
      </svg>
      <ul
        class="css-1a6m9ou-LineList exw3d030"
      >
        <li>
          <div
            class="css-1gmiakp-Line e16zi9h40"
          />
        </li>
        <li>
          <div
            class="css-1gmiakp-Line e16zi9h40"
          />
        </li>
        <li>
          <div
            class="css-1gmiakp-Line e16zi9h40"
          />
        </li>
        <li>
          <div
            class="css-1gmiakp-Line e16zi9h40"
          />
        </li>
      </ul>
    </div>
    <div
      class="css-q3qai6-StyledDiv e181dehd0"
    />
  </div>
</DocumentFragment>
`;

exports[`Skeleton renders correctly with type="line" 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    left: -25%;
  }

  to {
    left: 100%;
  }
}

.css-bgbcvp-StyledContainer {
  position: relative;
  display: block;
  width: 100%;
  overflow: hidden;
  cursor: progress;
}

.css-1gmiakp-Line {
  height: 12px;
  width: 120px;
  max-width: 100%;
  border-radius: 8px;
  background-color: #d4dae7;
}

.css-q3qai6-StyledDiv {
  position: absolute;
  top: 0;
  height: 100%;
  width: 25%;
  opacity: 0.8;
  background: linear-gradient(
      90deg,
      #ffffff00,
      #ffffff4D,
      #ffffff66,
      #ffffff4D,
      #ffffff00
    );
  -webkit-animation: animation-0 1s linear infinite;
  animation: animation-0 1s linear infinite;
  -webkit-animation-direction: alternate;
  animation-direction: alternate;
}

@media (prefers-reduced-motion: reduce) {
  .css-q3qai6-StyledDiv {
    -webkit-animation: unset;
    animation: unset;
  }
}

<div
    class="css-bgbcvp-StyledContainer e181dehd1"
  >
    <div
      class="css-1gmiakp-Line e16zi9h40"
    />
    <div
      class="css-q3qai6-StyledDiv e181dehd0"
    />
  </div>
</DocumentFragment>
`;

exports[`Skeleton renders correctly with type="list" 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    left: -25%;
  }

  to {
    left: 100%;
  }
}

.css-bgbcvp-StyledContainer {
  position: relative;
  display: block;
  width: 100%;
  overflow: hidden;
  cursor: progress;
}

.css-btk4pg-StyledList {
  list-style: none;
  padding: 0;
  margin: 0;
}

.css-btk4pg-StyledList>.e1550o002:nth-of-type(even) {
  background-color: #f6f6f8;
}

.css-1ldrf1m-StyledItem {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 48px;
  padding: 4px 8px;
}

.css-vmfh4y-StyledLine {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.css-1gmiakp-Line {
  height: 12px;
  width: 120px;
  max-width: 100%;
  border-radius: 8px;
  background-color: #d4dae7;
}

.css-q3qai6-StyledDiv {
  position: absolute;
  top: 0;
  height: 100%;
  width: 25%;
  opacity: 0.8;
  background: linear-gradient(
      90deg,
      #ffffff00,
      #ffffff4D,
      #ffffff66,
      #ffffff4D,
      #ffffff00
    );
  -webkit-animation: animation-0 1s linear infinite;
  animation: animation-0 1s linear infinite;
  -webkit-animation-direction: alternate;
  animation-direction: alternate;
}

@media (prefers-reduced-motion: reduce) {
  .css-q3qai6-StyledDiv {
    -webkit-animation: unset;
    animation: unset;
  }
}

<div
    class="css-bgbcvp-StyledContainer e181dehd1"
  >
    <ul
      class="css-btk4pg-StyledList e1550o000"
    >
      <li
        class="css-1ldrf1m-StyledItem e1550o002"
      >
        <div
          class="css-vmfh4y-StyledLine e1550o001"
        >
          <div
            class="css-1gmiakp-Line e16zi9h40"
          />
        </div>
        <div
          class="css-vmfh4y-StyledLine e1550o001"
        >
          <div
            class="css-1gmiakp-Line e16zi9h40"
          />
        </div>
        <div
          class="css-vmfh4y-StyledLine e1550o001"
        >
          <div
            class="css-1gmiakp-Line e16zi9h40"
          />
        </div>
      </li>
      <li
        class="css-1ldrf1m-StyledItem e1550o002"
      >
        <div
          class="css-vmfh4y-StyledLine e1550o001"
        >
          <div
            class="css-1gmiakp-Line e16zi9h40"
          />
        </div>
        <div
          class="css-vmfh4y-StyledLine e1550o001"
        >
          <div
            class="css-1gmiakp-Line e16zi9h40"
          />
        </div>
        <div
          class="css-vmfh4y-StyledLine e1550o001"
        >
          <div
            class="css-1gmiakp-Line e16zi9h40"
          />
        </div>
      </li>
      <li
        class="css-1ldrf1m-StyledItem e1550o002"
      >
        <div
          class="css-vmfh4y-StyledLine e1550o001"
        >
          <div
            class="css-1gmiakp-Line e16zi9h40"
          />
        </div>
        <div
          class="css-vmfh4y-StyledLine e1550o001"
        >
          <div
            class="css-1gmiakp-Line e16zi9h40"
          />
        </div>
        <div
          class="css-vmfh4y-StyledLine e1550o001"
        >
          <div
            class="css-1gmiakp-Line e16zi9h40"
          />
        </div>
      </li>
    </ul>
    <div
      class="css-q3qai6-StyledDiv e181dehd0"
    />
  </div>
</DocumentFragment>
`;

exports[`Skeleton renders correctly with type="slider" 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    left: -25%;
  }

  to {
    left: 100%;
  }
}

.css-bgbcvp-StyledContainer {
  position: relative;
  display: block;
  width: 100%;
  overflow: hidden;
  cursor: progress;
}

.css-1ibkrfq-StyledContainer {
  height: 277px;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: 16px;
  overflow: auto;
}

.css-1v2rm0w-StyledCard {
  border: 1px solid #d4dae7;
  border-radius: 4px;
  width: 239px;
  height: 261px;
  overflow: hidden;
}

.css-1qdcwka-StyledBanner {
  border: 1px solid #d4dae7;
  background-color: #d4dae7;
  width: 100%;
  height: 33%;
}

.css-q3qai6-StyledDiv {
  position: absolute;
  top: 0;
  height: 100%;
  width: 25%;
  opacity: 0.8;
  background: linear-gradient(
      90deg,
      #ffffff00,
      #ffffff4D,
      #ffffff66,
      #ffffff4D,
      #ffffff00
    );
  -webkit-animation: animation-0 1s linear infinite;
  animation: animation-0 1s linear infinite;
  -webkit-animation-direction: alternate;
  animation-direction: alternate;
}

@media (prefers-reduced-motion: reduce) {
  .css-q3qai6-StyledDiv {
    -webkit-animation: unset;
    animation: unset;
  }
}

<div
    class="css-bgbcvp-StyledContainer e181dehd1"
  >
    <div
      class="css-1ibkrfq-StyledContainer e10u8i0l2"
    >
      <div
        class="css-1v2rm0w-StyledCard e10u8i0l1"
      >
        <div
          class="css-1qdcwka-StyledBanner e10u8i0l0"
        />
      </div>
      <div
        class="css-1v2rm0w-StyledCard e10u8i0l1"
      >
        <div
          class="css-1qdcwka-StyledBanner e10u8i0l0"
        />
      </div>
      <div
        class="css-1v2rm0w-StyledCard e10u8i0l1"
      >
        <div
          class="css-1qdcwka-StyledBanner e10u8i0l0"
        />
      </div>
      <div
        class="css-1v2rm0w-StyledCard e10u8i0l1"
      >
        <div
          class="css-1qdcwka-StyledBanner e10u8i0l0"
        />
      </div>
    </div>
    <div
      class="css-q3qai6-StyledDiv e181dehd0"
    />
  </div>
</DocumentFragment>
`;

exports[`Skeleton renders default variant 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from {
    left: -25%;
  }

  to {
    left: 100%;
  }
}

.css-bgbcvp-StyledContainer {
  position: relative;
  display: block;
  width: 100%;
  overflow: hidden;
  cursor: progress;
}

.css-sprtbs-StyledContainer {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: 16px;
}

.css-2szx0p-Block {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 16px;
  border: 1px solid #d4dae7;
  border-radius: 4px;
  overflow: hidden;
}

.css-115kc0m-IconSkeleton {
  margin-right: 8px;
  width: 32px;
  height: 32px;
  min-width: 32px;
  border-radius: 8px;
  background-color: #d4dae7;
}

.css-1gmiakp-Line {
  height: 12px;
  width: 120px;
  max-width: 100%;
  border-radius: 8px;
  background-color: #d4dae7;
}

.css-q3qai6-StyledDiv {
  position: absolute;
  top: 0;
  height: 100%;
  width: 25%;
  opacity: 0.8;
  background: linear-gradient(
      90deg,
      #ffffff00,
      #ffffff4D,
      #ffffff66,
      #ffffff4D,
      #ffffff00
    );
  -webkit-animation: animation-0 1s linear infinite;
  animation: animation-0 1s linear infinite;
  -webkit-animation-direction: alternate;
  animation-direction: alternate;
}

@media (prefers-reduced-motion: reduce) {
  .css-q3qai6-StyledDiv {
    -webkit-animation: unset;
    animation: unset;
  }
}

<div
    class="css-bgbcvp-StyledContainer e181dehd1"
  >
    <div
      class="css-sprtbs-StyledContainer ellyxki1"
    >
      <div
        class="css-2szx0p-Block ellyxki0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-2szx0p-Block ellyxki0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-2szx0p-Block ellyxki0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-2szx0p-Block ellyxki0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-2szx0p-Block ellyxki0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-2szx0p-Block ellyxki0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-2szx0p-Block ellyxki0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
      <div
        class="css-2szx0p-Block ellyxki0"
      >
        <div
          class="css-115kc0m-IconSkeleton e19udb630"
        />
        <div
          class="css-1gmiakp-Line e16zi9h40"
        />
      </div>
    </div>
    <div
      class="css-q3qai6-StyledDiv e181dehd0"
    />
  </div>
</DocumentFragment>
`;
